src/espidf.rs

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 Developers of the Rand project.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Implementation for ESP-IDF
+use crate::Error;
+use core::ffi::c_void;
+
+extern "C" {
+    fn esp_fill_random(buf: *mut c_void, len: usize) -> u32;
+}
+
+pub fn getrandom_inner(dest: &mut [u8]) -> Result<(), Error> {
+    // Not that NOT enabling WiFi, BT, or the voltage noise entropy source (via `bootloader_random_enable`)
+    // will cause ESP-IDF to return pseudo-random numbers based on the voltage noise entropy, after the initial boot process:
+    // https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/random.html
+    //
+    // However tracking if some of these entropy sources is enabled is way too difficult to implement here
+    //
+    // Using esp_fill_random since it has some optimizations regarding filling a byte array from an
+    // u32 source.  See https://github.com/espressif/esp-idf/blob/master/components/esp_hw_support/hw_random.c
```

Comment:
These two lines are misleading in my opinion. There are not optimizations in the code itself, compiler is simply able to inline implementation of `esp_random` during compilation of the library. Also this optimization is still possible to perform with enabled LTO. So I think it's better to simply remove these lines.

Reply:
Ok, so just remove lines 22-25?

Reply:
Only 24-25.

Reply:
Done